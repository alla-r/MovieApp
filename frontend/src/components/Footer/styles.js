import styled from 'styled-components';

export const Background = styled.div`
  width: 100%;
  height: 160px;
  background: ${(props) => props.theme.colors.dark};
  @media only screen and (max-width: 768px) {
    height: 180px;
  }
`;

export const Container = styled.div`
  margin: 0 auto;
  width: 90%;
  max-width: 1320px;
  height: 100%;
  display: flex;
  flex-direction: row;
  justify-content: center;
  align-items: center;
  @media only screen and (max-width: 768px) {
    flex-direction: column;
  }
`;

export const Logo = styled.img`
  height: 20px;
  @media only screen and (max-width: 768px) {
    height: 18px;
  }
`;

export const Content = styled.p`
  margin-left: 30px;
  font-weight: 500;
  font-size: 16px;
  line-height: 24px;
  color: ${(props) => props.theme.colors.light};
  @media only screen and (max-width: 768px) {
    margin-left: 0px;
    margin-top: 30px;
    text-align: center;
    font-size: 14px;
    line-height: 22px;
  }
`;
