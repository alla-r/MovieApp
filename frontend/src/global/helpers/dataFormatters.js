/* eslint-disable camelcase */
const getFormattedPersonItem = (item) => {
  const formattedItem = {
    id: item.id,
    job: item.job || '',
    department: item.department || item.known_for_department,
    name: item.name,
    character: item.character,
    poster: `https://image.tmdb.org/t/p/original${item.profile_path}`,
  };

  return formattedItem;
};

const getFormattedRecommendationItem = (item) => {
  const formattedItem = {
    id: item.id,
    type: item.media_type,
    title: item.title || item.name,
    overview: item.overview,
    voteAvg: item.vote_average,
    date: item.release_date || item.first_air_date,
    poster: `https://image.tmdb.org/t/p/original${item.profile_path}`,
    backdrop: `https://image.tmdb.org/t/p/original${item.backdrop_path}`,
  };

  return formattedItem;
};

const getFormattedMediaDetails = (type, data) => {
  const formattedData = {};

  if (type === 'movie') {
    formattedData.details = {
      type,
      id: data.id,
      title: data.title,
      originalTitle: data.original_title,
      genres: data.genres,
      overview: data.overview,
      voteAvg: data.vote_average,
      status: data.status,
      poster: `https://image.tmdb.org/t/p/original${data.poster_path}`,
      backdrop: `https://image.tmdb.org/t/p/original${data.backdrop_path}`,
      productionCompanies: data.production_companies.map((item) => item.name),
      productionCountries: data.production_countries.map((item) => item.name),
      duration: data.runtime,
      revenue: data.revenue,
      budget: data.budget,
      date: data.release_date,
    };
    formattedData.cast = data.credits.cast.map((item) => getFormattedPersonItem(item));
    formattedData.crew = data.credits.crew.map((item) => getFormattedPersonItem(item));
    formattedData.recommendations = data.recommendations.results.map((item) =>
      getFormattedRecommendationItem(item),
    );
  }

  if (type === 'tv') {
    formattedData.details = {
      type,
      id: data.id,
      title: data.name,
      originalTitle: data.original_name,
      genres: data.genres,
      overview: data.overview,
      voteAvg: data.vote_average,
      status: data.status,
      poster: `https://image.tmdb.org/t/p/original${data.poster_path}`,
      backdrop: `https://image.tmdb.org/t/p/original${data.backdrop_path}`,
      productionCompanies: data.production_companies.map((item) => item.name),
      productionCountries: data.production_countries.map((item) => item.name),
      createdBy: data.created_by.map((creator) => ({
        id: creator.id,
        name: creator.name,
        job: 'Created By',
      })),
      numberOfSeasons: data.number_of_seasons,
      duration: data.episode_run_time[0],
      firstAirDate: data.first_air_date,
      lastAirDate: data.last_air_date,
    };
    formattedData.cast = data.credits.cast.map((item) => getFormattedPersonItem(item));
    formattedData.crew = data.credits.crew.map((item) => getFormattedPersonItem(item));
    formattedData.recommendations = data.recommendations.results.map((item) =>
      getFormattedRecommendationItem(item),
    );
  }

  return formattedData;
};

const getAge = (birthday, deathday) => {
  let age;
  let ageDifMs;
  let ageDate;

  if (deathday) {
    ageDifMs = new Date(deathday).getTime() - new Date(birthday).getTime();
    ageDate = new Date(ageDifMs);

    age = Math.abs(ageDate.getUTCFullYear() - 1970);
  } else {
    ageDifMs = new Date() - new Date(birthday).getTime();
    ageDate = new Date(ageDifMs);

    age = Math.abs(ageDate.getUTCFullYear() - 1970);
  }

  return age;
};

const getFormattedPersonDetails = (data = {}) => {
  const formattedDetails = {
    id: data.id,
    name: data.name,
    knownFor: data.known_for_department,
    birthday: `${data.birthday} (${getAge(data.birthday, data.deathday)} years old)`,
    deathday: data.deathday,
    placeOfBirth: data.place_of_birth,
    poster: `https://image.tmdb.org/t/p/original${data.profile_path}`,
    biography: data.biography,
    age: getAge(data.birthday, data.deathday),
  };

  return formattedDetails;
};

const getFormattedGenreList = (genreList, selectedGenresArr) => {
  const formattedGenreList = genreList.map(({ id, name }) => ({
    id,
    name,
    isChosen: selectedGenresArr.includes(id.toString()),
  }));

  return formattedGenreList;
};

const getFormattedItem = (item, type) => {
  const formattedItem = {
    id: item.id,
    type: item.media_type || type,
    date: item.release_date || item.first_air_date || 'unknown',
    title: item.title || item.name,
    description: item.overview,
    voteAvg: item.vote_average || 0,
    poster: `https://image.tmdb.org/t/p/original${item.poster_path}`,
  };

  return formattedItem;
};

const getFormattedListData = ({ page, total_pages, results }, mediaType) => {
  const formattedResultsList =
    results.length > 0 ? results.map((item) => getFormattedItem(item, mediaType)) : results;

  const formattedData = {
    page,
    totalPages: total_pages,
    results: formattedResultsList,
  };

  return formattedData;
};

const getFormattedSearchItem = (item, type) => {
  const formattedItem = {
    id: item.id,
    type: item.media_type || type,
    date: item.release_date || item.first_air_date || 'unknown',
    title: item.title || item.name,
    overview: item.overview,
    poster: `https://image.tmdb.org/t/p/original${item.poster_path}`,
  };

  return formattedItem;
};

const getFormattedSearchData = ({ page, total_pages, total_results, results }, mediaType) => {
  let formattedResultsList;

  if (mediaType === 'people' || mediaType === 'person') {
    formattedResultsList = results.map(getFormattedPersonItem);
  } else {
    formattedResultsList = results.map((item) => getFormattedSearchItem(item, mediaType));
  }

  const formattedData = {
    currentPage: page,
    totalPages: total_pages,
    results: formattedResultsList,
    totalResults: total_results,
  };

  return formattedData;
};

const getFormattedSocialMedia = ({ instagram_id, facebook_id, twitter_id }) => ({
  instagram: instagram_id ? `https://instagram.com/${instagram_id}` : null,
  facebook: facebook_id ? `https://www.facebook.com/${facebook_id}` : null,
  twitter: twitter_id ? `https://twitter.com/${twitter_id}` : null,
});

export {
  getFormattedMediaDetails,
  getFormattedGenreList,
  getFormattedListData,
  getFormattedSearchData,
  getFormattedPersonDetails,
  getFormattedSocialMedia,
};
