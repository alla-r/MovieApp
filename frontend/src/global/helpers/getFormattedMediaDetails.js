const getFormattedCreditsItem = (item) => {
  const formattedItem = {
    id: item.id,
    job: item.job || '',
    department: item.department || item.known_for_department,
    name: item.name,
    character: item.character,
    poster: `https://image.tmdb.org/t/p/original${item.profile_path}`,
  };

  return formattedItem;
};

const getFormattedRecommendationItem = (item) => {
  const formattedItem = {
    id: item.id,
    type: item.media_type,
    title: item.title || item.name,
    overview: item.overview,
    voteAvg: item.vote_average,
    date: item.release_date || item.first_air_date,
    poster: `https://image.tmdb.org/t/p/original${item.profile_path}`,
    backdrop: `https://image.tmdb.org/t/p/original${item.backdrop_path}`,
  };

  return formattedItem;
};

const getFormattedMediaDetails = (type, data) => {
  const formattedData = {};
  // console.log(type);
  // console.log(data);

  if (type === 'movie') {
    formattedData.details = {
      type,
      id: data.id,
      title: data.title,
      originalTitle: data.original_title,
      genres: data.genres,
      overview: data.overview,
      voteAvg: data.vote_average,
      status: data.status,
      poster: `https://image.tmdb.org/t/p/original${data.poster_path}`,
      backdrop: `https://image.tmdb.org/t/p/original${data.backdrop_path}`,
      productionCompanies: data.production_companies.map((item) => item.name),
      productionCountries: data.production_countries.map((item) => item.name),
      duration: data.runtime,
      revenue: data.revenue,
      budget: data.budget,
      date: data.release_date,
    };
    formattedData.cast = data.credits.cast.map((item) => getFormattedCreditsItem(item));
    formattedData.crew = data.credits.crew.map((item) => getFormattedCreditsItem(item));
    formattedData.recommendations = data.recommendations.results.map((item) =>
      getFormattedRecommendationItem(item),
    );
  }

  if (type === 'tv') {
    formattedData.details = {
      type,
      id: data.id,
      title: data.name,
      originalTitle: data.original_name,
      genres: data.genres,
      overview: data.overview,
      voteAvg: data.vote_average,
      status: data.status,
      poster: `https://image.tmdb.org/t/p/original${data.poster_path}`,
      backdrop: `https://image.tmdb.org/t/p/original${data.backdrop_path}`,
      productionCompanies: data.production_companies.map((item) => item.name),
      productionCountries: data.production_countries.map((item) => item.name),
      createdBy: data.created_by.map((creator) => ({
        id: creator.id,
        name: creator.name,
        job: 'Created By',
      })),
      numberOfSeasons: data.number_of_seasons,
      duration: data.episode_run_time[0],
      firstAirDate: data.first_air_date,
      lastAirDate: data.last_air_date,
    };
    formattedData.cast = data.credits.cast.map((item) => getFormattedCreditsItem(item));
    formattedData.crew = data.credits.crew.map((item) => getFormattedCreditsItem(item));
    formattedData.recommendations = data.recommendations.results.map((item) =>
      getFormattedRecommendationItem(item),
    );
  }

  return formattedData;
};

export default getFormattedMediaDetails;
