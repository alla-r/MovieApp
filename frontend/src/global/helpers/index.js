import {
  getFormattedMediaDetails,
  getFormattedGenreList,
  getFormattedListData,
} from './dataFormatters';
import ScrollToTop from './ScrollToTop';

export { getFormattedMediaDetails, ScrollToTop, getFormattedGenreList, getFormattedListData };

export const getFormattedItem = (item) => {
  const formattedItem = {
    id: item.id,
    type: item.media_type,
    date: item.release_date || item.first_air_date,
    title: item.title || item.name,
    description: item.overview,
    voteAvg: item.vote_average,
    poster: `https://image.tmdb.org/t/p/original${item.poster_path}`,
  };

  return formattedItem;
};

const BAR_COLORS = [
  {
    minValue: 70,
    maxValue: 100,
    path: 'rgba(141, 255, 71, 1)',
    trail: 'rgba(141, 255, 71, 0.3)',
  },
  {
    minValue: 30,
    maxValue: 69,
    path: 'rgba(255, 203, 71, 1)',
    trail: 'rgba(255, 203, 71, 0.3)',
  },
  {
    minValue: 0,
    maxValue: 29,
    path: 'rgba(255, 71, 71, 1)',
    trail: 'rgba(255, 71, 71, 0.3)',
  },
];

export const getBarColor = (type, percentageValue) => {
  const colors = BAR_COLORS.find(
    (element) => percentageValue >= element.minValue && percentageValue <= element.maxValue,
  );

  return (colors && colors[type]) || '#5B5B5B';
};

export const getFormattedDate = (dateString) => {
  const dateArray = dateString.split('-');
  const year = dateArray[0];
  const month = dateArray[1];
  const day = dateArray[2];
  const months = [
    'January',
    'February',
    'March',
    'April',
    'May',
    'June',
    'July',
    'August',
    'September',
    'October',
    'November',
    'December',
  ];

  return dateString ? `${months[month - 1].slice(0, 3)} ${day}, ${year}` : "";
};

export const getPercentageValue = (value, maxValue) => (value * 100) / maxValue;

export const getCrewListWithUniqueItems = (list) => {
  const names = new Set(list.map(({ name }) => name));
  const uniqueList = [];

  // eslint-disable-next-line no-restricted-syntax
  for (const uniqueName of names) {
    const jobsArray = list.filter(({ name }) => name === uniqueName).map(({ job }) => job);
    const { id } = list.find(({ name }) => name === uniqueName);
    uniqueList.push({
      id,
      job: jobsArray.join(', '),
      name: uniqueName,
    });
  }

  return uniqueList;
};

export const getFormattedYears = ({ type, date, firstAirDate, lastAirDate }) => {
  let formattedDate;

  if (type === 'movie') {
    formattedDate = date ? date.slice(0, 4) : null;
  }

  if (type === 'tv') {
    const startYear = firstAirDate ? firstAirDate.slice(0, 4) : null;

    const endYear = lastAirDate ? lastAirDate.slice(0, 4) : '...';
    formattedDate = startYear ? `${startYear} - ${endYear}` : "";
  }

  return formattedDate;
};

export const getDuration = ({ duration }) => {
  const hours = Math.trunc(duration / 60);
  const minutes = duration % 60;

  let formattedDuration = '';

  if (hours) {
    formattedDuration += `${hours}h `;
  }

  if (minutes) {
    formattedDuration += `${minutes}m`;
  }

  return formattedDuration;
};
