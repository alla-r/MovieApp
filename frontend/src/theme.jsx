const theme = {
  colors: {
    primary: '#D6535D',
    secondary: '#85CAD4',
    dark: '#1A191C',
    light: '#EAEAEA',
  },
};

export default theme;
