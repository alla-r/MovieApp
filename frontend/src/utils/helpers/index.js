import {
  getFormattedMediaDetails,
  getFormattedGenreList,
  getFormattedListData,
  getFormattedSearchData,
  getFormattedPersonDetails,
  getFormattedSocialMedia,
  getFormattedRecommendationItem,
} from './dataFormatters';

import {
  getFormattedItem,
  getBarColor,
  getFormattedDate,
  getPercentageValue,
  getCrewListWithUniqueItems,
  getFormattedYears,
  getDuration,
  getUniqueValues,
  showNotification,
  getTokenDuration,
} from './general';

import ScrollToTop from './ScrollToTop';

export {
  getFormattedMediaDetails,
  getFormattedGenreList,
  getFormattedListData,
  getFormattedSearchData,
  getFormattedPersonDetails,
  getFormattedSocialMedia,
  getFormattedRecommendationItem,
  getFormattedItem,
  getBarColor,
  getFormattedDate,
  getPercentageValue,
  getCrewListWithUniqueItems,
  getFormattedYears,
  getDuration,
  getUniqueValues,
  showNotification,
  getTokenDuration,
  ScrollToTop,
};
