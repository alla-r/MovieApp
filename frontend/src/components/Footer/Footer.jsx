import React from 'react';
import TMDbIcon from '../../global/images/Footer/tmdb_logo.svg';
import { Background, Container, Logo, Content } from './styles';

function Footer() {
  return (
    <Background>
      <Container>
        <Logo src={TMDbIcon} />
        <Content>This product uses the TMDB API but is not endorsed or certified by TMDB.</Content>
      </Container>
    </Background>
  );
}

export default Footer;
