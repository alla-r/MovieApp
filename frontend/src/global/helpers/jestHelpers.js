/* eslint-disable react/react-in-jsx-scope */
/* eslint-disable react/jsx-filename-extension */
/* eslint-disable import/prefer-default-export */
import { ThemeProvider } from 'styled-components';
import theme from '../../theme';

export const wrapperWithTheme = (fn, tree) => fn(
  <ThemeProvider theme={theme}>
    {tree}
  </ThemeProvider>
)