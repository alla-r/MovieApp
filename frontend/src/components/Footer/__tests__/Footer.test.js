import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import 'jest-styled-components';
import Footer from '../Footer';
import theme from '../../../theme';
import { wrapperWithTheme } from '../../../global/helpers';

describe('<Footer />', () => {
  const component = () => wrapperWithTheme(shallow, <Footer />);
  
  it('should be defined', () => {
    const wrapper = component();

    expect(wrapper).toBeDefined();
  });

  it('should match snapshot', () => {
    const wrapper = component();
    const tree = renderer.create(wrapper).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
