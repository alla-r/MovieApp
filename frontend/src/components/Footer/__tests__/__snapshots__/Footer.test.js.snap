// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer /> should match snapshot 1`] = `
.c0 {
  width: 100%;
  height: 160px;
  background: #1A191C;
}

.c1 {
  margin: 0 auto;
  width: 90%;
  max-width: 1320px;
  height: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c2 {
  height: 20px;
}

.c3 {
  margin-left: 30px;
  font-weight: 500;
  font-size: 16px;
  line-height: 24px;
  color: #EAEAEA;
}

@media only screen and (max-width:768px) {
  .c0 {
    height: 180px;
  }
}

@media only screen and (max-width:768px) {
  .c1 {
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
  }
}

@media only screen and (max-width:768px) {
  .c2 {
    height: 18px;
  }
}

@media only screen and (max-width:768px) {
  .c3 {
    margin-left: 0px;
    margin-top: 30px;
    text-align: center;
    font-size: 14px;
    line-height: 22px;
  }
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    <img
      className="c2"
      src="tmdb_logo.svg"
    />
    <p
      className="c3"
    >
      This product uses the TMDB API but is not endorsed or certified by TMDB.
    </p>
  </div>
</div>
`;
